by-language/java-qa/src/test/java/io/crate/qa/JdbcMetaDataTest.java

```diff
@@ -0,0 +1,1238 @@
+package stock_jdbc;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.RowIdLifetime;
+import java.util.Map;
+
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import io.crate.testing.CrateTestCluster;
+
+@RunWith(JUnit4.class)
+public class JdbcMetaDataTest {
+
+    @ClassRule
+    public static final CrateTestCluster TEST_CLUSTER = CrateTestCluster
+        .fromURL("https://cdn.crate.io/downloads/releases/nightly/crate-latest.tar.gz")
+        .settings(Map.of("psql.port", 55432))
+        .build();
+    public static final String URL = "jdbc:postgresql://localhost:55432/doc?user=crate";
+
+    @Test
+    public void test_allProceduresAreCallable() throws Exception {
+        try (var conn = DriverManager.getConnection(URL)) {
+            assertThat(conn.getMetaData().allProceduresAreCallable(), is(true));
+        }
+    }
+
+    @Test
+    public void test_allTablesAreSelectable() throws Exception {
+        try (var conn = DriverManager.getConnection(URL)) {
+            assertThat(conn.getMetaData().allTablesAreSelectable(), is(true));
+        }
+    }
+
+    @Test
+    public void test_autoCommitFailureClosesAllResultSets() throws Exception {
+        try (var conn = DriverManager.getConnection(URL)) {
+            assertThat(conn.getMetaData().autoCommitFailureClosesAllResultSets(), is(false));
+        }
+    }
+
+    @Test
+    public void test_dataDefinitionCausesTransactionCommit_TODO() throws Exception {
```

Comment:
Please remove `TODO`, change the assert to true and add an `@Ignore` annotation, with a description, It will be much more visible, that it's not supported, and easier to find in the future if we ever support commit (Same as `test_getAttributes()`)

Reply:
Adjusted with c08fc31dbbbd.